consumer/src/main/java/com/flipkart/varadhi/consumer/concurrent/EventExecutorGroup.java

```diff
@@ -0,0 +1,95 @@
+package com.flipkart.varadhi.consumer.concurrent;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.*;
+
+public class EventExecutorGroup implements ScheduledExecutorService {
```

Comment:
what the purpose of this?

Reply:
It's not used for now, but will be used in future commits. This will represent a ThreadPool which are collectively "consuming" messages for many subscriptions. This will also provide us a way to "schedule" events.

In java terms, ThreadPool contains many threads.
In netty terms, Event(Loop/Executor)Group contains many EventExecutors.

The difference in naming is significant. Thread is just that, a simple "thread". But an EventExecutor is actually a (thread + a task queue). Essentially EventExecutor == java's ExecutorService with only 1 thread.